include/openthread/channel_manager.h

```diff
@@ -76,6 +79,7 @@ void otChannelManagerRequestChannelChange(otInstance *aInstance, uint8_t aChanne
  */
 uint8_t otChannelManagerGetRequestedChannel(otInstance *aInstance);
 
+#if OPENTHREAD_FTD
```

Comment:
We cannot add `#if OPENTHREAD_FTD` (or check for any other OT configs) in any of the public/platform header files since these headers may be included in other codes that intend to use OT API where these definitions/configs may not be available or defined.

The solution is to document that certain API requires certain config in the header file
And then provide the API implementation only under that `#if ` check

Reply:
Thx, of course.